api/chalicelib/core/feature_flags.py

```diff
@@ -0,0 +1,347 @@
+    to_be_deleted = []
+    to_be_updated = []
+    to_be_created = []
+
+    for cid in existing_ids:
+        if cid not in [c.condition_id for c in conditions]:
+            to_be_deleted.append(cid)
+
+    for condition in conditions:
+        if condition.condition_id is None:
+            to_be_created.append(condition)
+        else:
+            to_be_updated.append(condition)
+
+    if len(to_be_created) > 0:
+        create_conditions(feature_flag_id=feature_flag_id, conditions=to_be_created)
+
+    if len(to_be_updated) > 0:
+        update_conditions(feature_flag_id=feature_flag_id, conditions=to_be_updated)
+
+    if len(to_be_deleted) > 0:
+        delete_conditions(feature_flag_id=feature_flag_id, ids=to_be_deleted)
+
+    return get_conditions(feature_flag_id)
+
+
+def update_conditions(feature_flag_id: int, conditions: List[schemas.FeatureFlagCondition]) -> Any:
+    """
+    Update existing feature flag conditions and return their updated data.
+    """
+    values = []
+    params = {
+        "feature_flag_id": feature_flag_id,
+    }
+    for i in range(len(conditions)):
+        values.append(f"(%(condition_id_{i})s, %(name_{i})s, %(rollout_percentage_{i})s, %(filters_{i})s::jsonb)")
+        params[f"condition_id_{i}"] = conditions[i].condition_id
+        params[f"name_{i}"] = conditions[i].name
+        params[f"rollout_percentage_{i}"] = conditions[i].rollout_percentage
+        params[f"filters_{i}"] = json.dumps(conditions[i].filters)
+
+    sql = f"""
+        UPDATE feature_flags_conditions
+        SET name = c.name, rollout_percentage = c.rollout_percentage, filters = c.filters
+        FROM (VALUES {','.join(values)}) AS c(condition_id, name, rollout_percentage, filters)
+        WHERE c.condition_id = feature_flags_conditions.condition_id AND feature_flag_id = %(feature_flag_id)s;
+    """
+
+    with pg_client.PostgresClient() as cur:
+        query = cur.mogrify(sql, params)
+        cur.execute(query)
+
+
+def delete_conditions(feature_flag_id: int, ids: List[int]) -> None:
+    """
+    Delete feature flag conditions.
+    """
+    sql = """
+        DELETE FROM feature_flags_conditions
+        WHERE condition_id IN %(ids)s
+            AND feature_flag_id= %(feature_flag_id)s;
+    """
+
+    with pg_client.PostgresClient() as cur:
+        query = cur.mogrify(sql, {"feature_flag_id": feature_flag_id, "ids": tuple(ids)})
+        cur.execute(query)
+
+
+def delete_feature_flag(project_id: int, feature_flag_id: int):
+    """
+    Delete a feature flag.
+    """
+    conditions = [
+        "project_id=%(project_id)s",
+        "feature_flags.feature_flag_id=%(feature_flag_id)s"
+    ]
+    params = {"project_id": project_id, "feature_flag_id": feature_flag_id}
+    with pg_client.PostgresClient() as cur:
+        query = cur.mogrify(f"""UPDATE feature_flags
+                                SET deleted_at= (now() at time zone 'utc')
```

Comment:
@shekarsiri Let's add `is_active = true` on delete action. It will help to select active and not deleted flags on backend side using only `is_active` field.